api/room/schema.py

```diff
@@ -88,6 +88,12 @@ def mutate(self, info, office_id, **kwargs):
 
         admin_roles.create_rooms_update_delete_office(office_id)
         query = Room.get_query(info)
+        query_result = query.filter(
+            RoomModel.calendar_id == kwargs.get('calendar_id'))
+        if query_result.count() > 0:
+            ErrorHandler.check_conflict(
+                self, kwargs['calendar_id'], 'CalenderId')
```

Comment:
For this I think you should query with the filter for state-active. The way you have implemented it someone may be prevented from creating a new room even though the original room with that calendar_is has been archived.

Reply:
thanx for pointing this out @caveinn. It has been worked on.